RPC config updates - Closes #1287 (#1317)
LiskHQ/lisk-docs: modules/ROOT/pages/guides/app-development/configuration.adoc

:url_guides_module: guides/app-development/module.adoc
:url_guides_plugin: guides/app-development/plugin.adoc
:url_guides_setup: guides/app-development/setup.adoc

> :url_guides_non_forging_pretection: guides/node-management/non-forging-node-protection.adoc

Author: ```suggestion
:url_guides_non_forging_protection: guides/node-management/non-forging-node-protection.adoc
```

How to configure a blockchain application.

The `Application` instance expects two arguments:

. A configuration file
. A genesis block

.src/app/index.js
[source,js]
----
import { Application, PartialApplicationConfig } from 'lisk-sdk';
import { registerModules } from './modules';
import { registerPlugins } from './plugins';

export const getApplication = (
	genesisBlock: Record<string, unknown>,
	config: PartialApplicationConfig,
): Application => {
	const app = Application.defaultApplication(genesisBlock, config);

	registerModules(app);
	registerPlugins(app);

	return app;
};
----

== Setting a custom configuration

To start the application with a custom `config.json`, add the `--config` flag to the xref:{url_references_cli_start}[start command] with a custom path to the configuration file, e.g. `--config my_custom_config.json`.

[source,bash]
----
bin/run start --config my_custom_config.json --override-config
----

=== How to create a custom config

> The best way to create a custom configuration for your app is to copy an existing `config.json` and adjust it to your needs.

Reviewer: ```suggestion
The optimal method to create a custom configuration for your app is to copy an existing `config.json` and adjust it to suit your requirements.
```

For example, copy the default `config.json` of the blockchain application:

.In the root directory of the blockchain application
[source,bash]
----
cp config/default/config.json my_custom_config.json

> ----

Reviewer 2: > copy an existing `config.json` and adjust it to suit your requirements.
Since you're giving the example to copy maybe also give an example to "adjust it" to complete the narrative

Author: I rearranged the sections accordingly.

== Config file location

> The currently used configuration of a blockchain application is stored under the following path, after the applications was started for the first time:

Reviewer: ```suggestion
The currently used configuration of a blockchain application is stored under the following path after the applications were started for the first time:
```

Author: ```suggestion
The currently used configuration of a blockchain application is stored under the following path after the application was started for the first time:
```

> The currently used configuration of a blockchain application is stored under the following path after the application was started for the first time:

Reviewer 2: ```suggestion
The current configuration of a blockchain application is stored under the following path after the application is started for the first time:
```

}
----

> Restart the app to apply the new configuration.

Reviewer 2: Looking at the flow, I think this come before the section "Config file location"

====
Allowing arbitrary machines to access the rpc port (ws) is dangerous and strongly discouraged -- access should be strictly limited to trusted machines.

> See for reference the guide xref:{url_guides_non_forging_pretection}[]

Reviewer: ```suggestion
For reference see the guide xref:{url_guides_non_forging_protection}[]
```

        "enable": true, //true or false

>         "mode": "ws", //"ws" or "ipc"

Reviewer 2: A short explainer b/w `ws` & `ipc`(when to use which) would be helpful here or is it already mentioned somewhere else ?

Author: Added the explanation to the communications page as discussed
